zingo-sync/src/sync.rs

```diff
@@ -27,62 +31,107 @@ pub async fn sync<P, W>(
     wallet: &mut W,
 ) -> Result<(), ()>
 where
-    P: Parameters + Send + 'static,
-    W: SyncWallet,
+    P: Parameters + Sync + Send + 'static,
+    W: SyncWallet + SyncBlocks + SyncNullifiers + SyncShardTrees,
 {
     tracing::info!("Syncing wallet...");
 
-    // TODO: add trait methods to read/write wallet data to/from sync engine
-    // this is where sync state would be read from wallet data
-    let sync_state = SyncState::new(); // placeholder
+    let mut handles = Vec::new();
 
     // create channel for sending fetch requests and launch fetcher task
-    let (fetch_request_sender, fetch_request_receiver) = unbounded_channel();
+    let (fetch_request_sender, fetch_request_receiver) = mpsc::unbounded_channel();
```

Comment:
because, only because i have semantic saturation on the word 'fetch', this bit could use commentary